src/rebar3_hex_cut.erl

```diff
@@ -24,116 +24,134 @@ init(State) ->
+    NewVersion = increment_version(Type, ResolvedVersion),
     AppSrcFile = rebar_app_info:app_file_src(App),
 
     case Version of
         _Git when Version =:= git orelse Version =:= "git" ->
-            rebar_api:info("Creating new tag v~s...", [NewVersion]),
-            rebar_utils:sh(io_lib:format("git tag v~s", [NewVersion]), []),
-
-            {application, _, AppDetails} = rebar3_hex_file:update_app_src(App, NewVersion),
-
-            case rebar3_hex_publish:validate_app_details(AppDetails) of
-                ok ->
-                    Name = rebar_app_info:name(App),
-                    Deps = rebar_state:get(State, {locks, default}, []),
-                    {TopLevel, Excluded} = rebar3_hex_publish:gather_deps(Deps),
-                    case rebar3_hex_publish:publish(App, Name, NewVersion, TopLevel,
-                                                    Excluded, AppDetails, HexConfig, State) of
-                        {ok, _State} ->
-                            case rebar3_hex_io:ask("Push new tag to origin?", boolean, "Y") of
-                                true ->
-                                    rebar_api:info("Pushing new tag v~s...", [NewVersion]),
-                                    rebar_utils:sh(io_lib:format("git push origin v~s", [NewVersion]), []),
-                                    {ok, State};
-                                false ->
-                                    {ok, State}
-                            end;
-                        {error, _} = Error ->
-                            rebar_api:info("Deleting new tag v~s...", [NewVersion]),
-                            rebar_utils:sh(io_lib:format("git tag -d v~s", [NewVersion]), []),
-                            Error
+            create_tag(NewVersion),
+
+            case try_publish(State, Repo, App, Args) of
+                {ok, _State} ->
+                    case rebar3_hex_io:ask("Push new tag to origin?", boolean, "Y") of
+                        true ->
+                            maybe_push_tag(Version),
+                            {ok, State};
+                        false ->
+                            {ok, State}
                     end;
-                Error ->
-                    Error
+                _ ->
+                    delete_tag(NewVersion),
+                   {ok, State}
             end;
         _ ->
             Spec = rebar3_hex_file:update_app_src(App, NewVersion),
             NewAppSrcFile = io_lib:format("~tp.\n", [Spec]),
             ok = rebar_file_utils:write_file_if_contents_differ(AppSrcFile, NewAppSrcFile),
             ask_commit_and_push(NewVersion),
-            case rebar3_hex_publish:publish(rebar_app_info:original_vsn(App, NewVersion), HexConfig, State) of
-                {ok, _State} ->
+            case try_publish(State, Repo, rebar_app_info:original_vsn(App, NewVersion), Args) of
+                {ok, _} ->
+                    maybe_create_and_push_tag(NewVersion),
```

Comment:
This one is new. Before we were not offering to create and push and a tag for this case... maybe we shouldn't but I thought it made sense.